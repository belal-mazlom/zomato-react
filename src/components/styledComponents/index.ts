import { Link } from 'gatsby';
import styled, { createGlobalStyle } from 'styled-components';

//OpenSans-Regular.ttf

export const GlobalStyle = createGlobalStyle`
  @font-face {
    font-family: 'Open Sans';
    src: url(${require('@assets/fonts/OpenSans-Italic.ttf')}) format('truetype');
    font-weight: normal;
    font-style: italic;
  }
  @font-face {
    font-family: 'Open Sans';
    src: url(${require('@assets/fonts/OpenSans-Light.ttf')}) format('truetype');
    font-weight: 300;
    font-style: normal;
  }
  @font-face {
    font-family: 'Open Sans';
    src: url(${require('@assets/fonts/OpenSans-LightItalic.ttf')}) format('truetype');
    font-weight: 300;
    font-style: italic;
  }
  @font-face {
    font-family: 'Open Sans';
    src: url(${require('@assets/fonts/OpenSans-Regular.ttf')}) format('truetype');
    font-weight: 400;
    font-style: normal;
  }
  @font-face {
    font-family: 'Open Sans';
    src: url(${require('@assets/fonts/OpenSans-SemiBold.ttf')}) format('truetype');
    font-weight: 600;
    font-style: normal;
  }
  @font-face {
    font-family: 'Open Sans';
    src: url(${require('@assets/fonts/OpenSans-SemiBoldItalic.ttf')}) format('truetype');
    font-weight: 600;
    font-style: italic;
  }
  @font-face {
    font-family: 'Open Sans';
    src: url(${require('@assets/fonts/OpenSans-Bold.ttf')}) format('truetype');
    font-weight: 700;
    font-style: normal;
  }
  @font-face {
    font-family: 'Open Sans';
    src: url(${require('@assets/fonts/OpenSans-BoldItalic.ttf')}) format('truetype');
    font-weight: 700;
    font-style: italic;
  }
  @font-face {
    font-family: 'Open Sans';
    src: url(${require('@assets/fonts/OpenSans-ExtraBold.ttf')}) format('truetype');
    font-weight: 800;
    font-style: normal;
  }
  @font-face {
    font-family: 'Open Sans';
    src: url(${require('@assets/fonts/OpenSans-ExtraBoldItalic.ttf')}) format('truetype');
    font-weight: 800;
    font-style: italic;
  }
  html {
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  font: 62.5% open-sans;
  box-sizing: border-box;
  overflow-y: scroll;
  scroll-behavior: smooth;
}
body {
  box-sizing: border-box;
  background-color: ${props => props.theme.bgColor};
  color: hsla(0, 0%, 0%, 0.8);
  color: ${props => props.theme.txtColor};
  font-family: 'Open sans', sans-serif;
  font-weight: normal;
  font-kerning: normal;
  font-feature-settings: "kern", "liga", "clig", "calt";
  margin: 0;
  transition: all 0.25s linear;
  word-wrap: break-word;
  -moz-font-feature-settings: "kern", "liga", "clig", "calt";
  -ms-font-feature-settings: "kern", "liga", "clig", "calt";
  -webkit-font-feature-settings: "kern", "liga", "clig", "calt";
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}
`;

export const TopHeader = styled.header`
  align-items: center;
  background-color: ${props => props.theme.secondaryBgColor};
  display: flex;
  justify-content: space-between;
  padding: 0rem 1rem;
  height: 3.4rem;
  margin-top: 0rem;
  transition: margin-top 0.4s ease-in-out;
  &.on-top {
    margin-top: -17.2rem;
  }
`;

export const BackHome = styled(Link)`
  display: flex;
  text-decoration: none;
`;

export const Logo = styled.div.attrs({
  title: 'Logo',
})`
  background: url(${require('@images/zomato.svg')}) no-repeat 0 0;
  background-size: 9rem 3rem;
  height: 3rem;
  width: 9rem;
`;

export const LogoTitle = styled.span`
  font-size: 1.8rem;
  color: ${props => props.theme.txtColor};
  line-height: 3.1rem;
  margin: 0 1rem;
`;
