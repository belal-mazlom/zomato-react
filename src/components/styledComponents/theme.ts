import { createContext } from 'react';
import { DefaultTheme } from 'styled-components';

export const ThemeContext = createContext({
  toggleTheme: () => {},
});

export const defaultTheme: DefaultTheme = {
  primaryColor: '#50BEBF',
  secondaryColor: '#51BF56',
  txtColor: '#4e5456',
  grayColor: '#b2b6b7',
  darkColor: '#333333',
  darkGrayColor: '#777777',
  bgColor: '#F1F1F1',
  secondaryBgColor: '#E5E5E5',
  thirdBgColor: '#FFFFFF',
  sideBgColor: '#E5E5E5',
  borderColor: '#999999',
  white: '#FFFFFF',
  black: '#333333',
  red: '#bf5050',
};

export const darkTheme: DefaultTheme = {
  ...defaultTheme,
  txtColor: '#FFFFFF',
  bgColor: '#2b2b2b',
  secondaryBgColor: '#50BEBF',
  thirdBgColor: '#222222',
  sideBgColor: '#222222',
  grayColor: '#EEEEEE',
  darkColor: '#F1F1F1',
  darkGrayColor: '#E2E2E2',
};
