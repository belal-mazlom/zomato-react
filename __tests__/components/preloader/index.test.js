import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from '@components/styledComponents/theme';
import Preloader from '@components/preloader/index';

describe('<Preloader/>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <Preloader show={true} />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
