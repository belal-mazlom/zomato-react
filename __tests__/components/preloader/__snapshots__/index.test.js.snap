// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Preloader/> renders correctly 1`] = `
<div
  className="sc-bdfBwQ dXgGps"
>
  <div
    className="sc-gsTCUz hcrgbO"
  >
    <img
      src="test-file-stub"
    />
    <span>
      Loading ...
    </span>
  </div>
</div>
`;
